src/general/Jukebox.cs

```diff
@@ -556,6 +556,18 @@ private void PlayNextTrackFromList(TrackList list, Func<int, AudioPlayer> getPla
         if (tracks.Length == 0)
             return;
 
+        var random = new XoShiRo128starstar();
+
+        var contextMusicOnly = tracks.Where(c => c.ExclusiveToContexts != null).ToArray();
+
+        if (contextMusicOnly.Length > 0)
+        {
+            if (random.Next(0, Constants.CONTEXTUAL_MUSIC_RARITY) == 0)
+            {
```

Comment:
Would it be better to make the random check before getting the context tracks for some extra performance?

Reply:
Smart